### ISSUE TITLE
Internal error "assert(inList.length > 0)" stops analysing.

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#187]
**Describe the bug**
Seeing error message "An internal error occurred while performing type analysis" in one of my file.

**To Reproduce**
Not sure how to reproduce, but I'll provide full code of the file below.

**Expected behavior**

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/3479453/62754574-f9cde780-baa2-11e9-93b0-a96165953e1e.png)
_error hint_

![image](https://user-images.githubusercontent.com/3479453/62754577-ffc3c880-baa2-11e9-90f5-33d7ac4d22f0.png)
_output log_

full code of the file `cache_helper.py`
```
import datetime
import json

from app.base import redis
from app.models.user import User

EXPIRE_TIME_LONG = 7 * 24 * 3600
EXPIRE_TIME_NORMAL = 24 * 3600
EXPIRE_TIME_SHORT = 2 * 3600


def to_cache_dict(model, cache_keys=[]):
    #d = model.get_attr_dict()
    d = model.to_dict()
    cache_data = {}
    for key in cache_keys:
        cache_data[key] = d[key]
        if type(cache_data[key]) is datetime.datetime:
            cache_data[key] = cache_data[key].timestamp()
        if type(cache_data[key]) is datetime.date:
            _datetime = datetime.datetime.combine(cache_data[key], datetime.datetime.min.time())
            cache_data[key] = _datetime.timestamp()
    return cache_data


def get_user_cache(token):
    cache_hash = "user:%s" % token
    return redis.get(cache_hash)


def update_user_cache(user):
    if not user:
        return None
    cache_hash = "user:%s" % user.token
    cache_data = to_cache_dict(user, User.cache_keys)
    redis.set(cache_hash, json.dumps(cache_data))
    redis.expire(cache_hash, EXPIRE_TIME_NORMAL)
    return cache_data
```

**VS Code extension or command-line**
Using VSCode extension 1.0.53

**Additional context**
Not sure if there's somthing else relatied, please let me know if you need more information.

**Comment by erictraut** [comment]
Thanks for the bug report. I've isolated the cause of the internal error. This will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.54, which I just published.

**Comment by giyyapan** [comment]
Fix confirmed, thanks!
